File: Docs/CS/Data/Events/ChannelEvent.cs
using System;
using System.ComponentModel;
using System.Threading.Tasks;

namespace DevExpress.DevAV.Chat.Events {

    public abstract class ChannelEvent {
        protected ChannelEvent(IChannel channel) {
            this.Channel = channel;
        }
        public IChannel Channel {
            get;
        }
        public string UserName {
            get { return Channel.UserName; }
        }
    }
    //
    public class CredentialsRequiredEvent : ChannelEvent {
        public string Salt {
            get;
        }
        public void SetAccessTokenQuery(Task<string> query) {
            /* some code */
        }
        [EditorBrowsable(EditorBrowsableState.Never)]
        public Task<string> GetAccessTokenQuery() {
            /* some code*/
        }
    }
    public class ChannelReadyEvent : ChannelEvent {
        public ChannelReadyEvent(IChannel channel)
            : base(channel) {
        }
    }
}

File: Docs/VB/Data/Events/ChannelEvent.vb
Imports System
Imports System.ComponentModel
Imports System.Threading.Tasks

Namespace DevExpress.DevAV.Chat.Events

    Public MustInherit Class ChannelEvent
        Protected Sub New(ByVal channel As IChannel)
            Me.Channel = channel
        End Sub
        Public ReadOnly Property Channel() As IChannel
        Public ReadOnly Property UserName() As String
            Get
                Return Channel.UserName
            End Get
        End Property
    End Class
    '
    Public Class CredentialsRequiredEvent
        Inherits ChannelEvent

        Public ReadOnly Property Salt() As String
        Public Sub SetAccessTokenQuery(ByVal query As Task(Of String))
            ' some code 
        End Sub
        <EditorBrowsable(EditorBrowsableState.Never)>
        Public Function GetAccessTokenQuery() As Task(Of String)
            ' some code
        End Function
    End Class
    Public Class ChannelReadyEvent
        Inherits ChannelEvent

        Public Sub New(ByVal channel As IChannel)
            MyBase.New(channel)
        End Sub
    End Class
End Namespace